container-storage-setup.sh

```diff
@@ -437,6 +437,46 @@ check_docker_storage_metadata() {
   exit 1
 }
 
+# See whether we can use overlay backend
+can_do_overlay() {
+    if test -d $_DOCKER_METADATA_DIR; then
+        mountroot=$_DOCKER_METADATA_DIR
+    else
+        mountroot=/var
```

Comment:
This function can be useful both in compatibility mode as well as non-compatibility mode. So anything which ties it to compatibility mode should be outside the function. I think it is better to not check  `_DOCKER_METADATA_DIR` here. Instead caller should pass the mountroot argument to function.